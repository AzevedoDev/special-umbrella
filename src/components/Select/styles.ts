import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    max-width: 23rem;
    height: 5rem;
    color: ${theme.colors.black};
    background-color: ${theme.colors.white};
    border-radius: 0.6rem;
    box-shadow: 0 0.2rem 0.6rem rgba(0, 0, 0, 0.05);

    &:hover {
      box-shadow: #848484 0 0 0.1rem 0.1rem;
    }
    &:focus {
      box-shadow: #848484 0 0 0.1rem 0.1rem;
    }
    select {
      width: 100%;
      height: 100%;
      padding: 0 3.3rem;
      color: ${theme.colors.black};
      background-color: transparent;
      border: 0;
      appearance: none;
      outline: none;
      cursor: pointer;
      font-family: ${theme.font.family};
    }
    &::after {
      position: absolute;
      top: 2.2rem;
      right: 2rem;
      width: 1.1rem;
      height: 0.6rem;
      pointer-events: none;
      content: '';
      background-color: #848484;
      clip-path: polygon(100% 0%, 0 0%, 50% 100%);
    }
  `}
`
