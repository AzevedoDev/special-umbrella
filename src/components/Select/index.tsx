import * as S from './styles'

export type SelectProps = {
  lotteries: Loteria[]
}
type Loteria = {
  id: number
  nome: string
}
const Select = ({ lotteries }: SelectProps) => (
  <S.Wrapper>
    <select aria-label="select-lotteries">
      {lotteries.map(({ nome, id }) => (
        <option key={id} value={id}>
          {nome.toUpperCase()}
        </option>
      ))}
    </select>
  </S.Wrapper>
)

export default Select
