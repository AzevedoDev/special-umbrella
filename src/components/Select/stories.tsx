import { Story, Meta } from '@storybook/react/types-6-0'
import Select, { SelectProps } from '.'

export default {
  title: 'Select',
  component: Select,
  args: {
    lotteries: [
      {
        id: 0,
        nome: 'mega-sena'
      },
      {
        id: 1,
        nome: 'quina'
      }
    ]
  }
} as Meta

export const Default: Story<SelectProps> = (args) => <Select {...args} />
