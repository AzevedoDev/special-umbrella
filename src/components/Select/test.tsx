import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Select from '.'

describe('<Select />', () => {
  const lotteries = [
    {
      id: 0,
      nome: 'mega-sena'
    },
    {
      id: 1,
      nome: 'quina'
    }
  ]
  it('should render and match with snapshot', () => {
    renderWithTheme(<Select lotteries={lotteries} />)

    expect(screen.getByLabelText(/select-lotteries/i)).toBeInTheDocument()
    expect(screen.getByLabelText(/select-lotteries/i)).toMatchSnapshot()
  })
  it('should select a QUINA option', () => {
    renderWithTheme(<Select lotteries={lotteries} />)
    const select = screen.getByLabelText(/select-lotteries/i)
    select
    expect(screen.getByLabelText(/select-lotteries/i)).toBeInTheDocument()
  })
})
