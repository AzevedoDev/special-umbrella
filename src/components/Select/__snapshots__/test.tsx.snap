// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render and match with snapshot 1`] = `
<select
  aria-label="select-lotteries"
>
  <option
    value="0"
  >
    MEGA-SENA
  </option>
  <option
    value="1"
  >
    QUINA
  </option>
</select>
`
